### PR TITLE
Update for ark integration

### PR DESCRIPTION
**Pull request by yrong**
The initial intention is for migrating bls algorithm from milagro to [ark curves](https://github.com/arkworks-rs/curves). But for G1 point can't be scale encoded and cached into storage so the benchmark is worse than milagro. 

It's still in investigation and I'll come back when finished the oak issues.

### CODE REVIEW

File: parachain/primitives/beacon/src/ark_bls.rs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+}
+
+pub fn fast_aggregate_verify(
+	pub_keys: Vec<Vec<u8>>,
+	message: Vec<u8>,
+	signature: Vec<u8>,
+) -> Result<bool, BlsError> {
+	let sig = Signature::from_bytes(&signature).map_err(|_| BlsError::InvalidSignature)?;
+	let pub_keys: Result<Vec<PublicKey>, _> =
+		pub_keys.into_iter().map(|pk| PublicKey::from_bytes(&pk)).collect();
```
Comment (yrong) [inline parachain/primitives/beacon/src/ark_bls.rs:95]: The decompressed pubkey should be read from cached storage as we do for Milagro.

**Comment by claravanstaden** [comment]
Cool investigation! Why can't G1 point be scale encoded?

File: parachain/primitives/beacon/src/ark_bls.rs
Comment (vgeddes) [inline parachain/primitives/beacon/src/ark_bls.rs:1]: Hey, curious where you got this code. 

Did you see this comment from Seun by the way? It's another BLS lib based on Arkworks.

https://matrix.to/#/!dOiGCcBecwSGXtifov:web3.foundation/$169358770621189BwbgH:matrix.org?via=web3.foundation&via=matrix.org&via=matrix.parity.io

Comment (yrong) [reply parachain/primitives/beacon/src/ark_bls.rs:1]: No, it's also from other projects. I've checked the implementation a bit and seems the logic/library used underying almost the same. And I would assume what we need is only the primitives for `fast_aggregate_verify` so a thin wrapper of the Arkworks library should be enough.

**Pull request closed without merging**